scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -83,9 +85,13 @@ class AsyncFlowDefRunner extends Writer { self =>
 
   private[this] val mutex = new AnyRef
 
+  type StateKey[T] = (Config, Mode, TypedPipe[T])
+  type WorkVal[T] = Future[TypedPipe[T]]
+
   private case class State(
     filesToCleanup: Map[Mode, Set[String]],
-    forcedPipes: Map[(Config, Mode, TypedPipe[Any]), Future[TypedPipe[Any]]]) {
+    initToOpt: HMap[TypedPipe, TypedPipe],
+    forcedPipes: HMap[StateKey, WorkVal]) {
```

Comment:
probably worth ensuring we have good comments on all the state fields here (i know this is old, but helps review/reading). i.e. does force mean the same as in terms of the TypedPipe force to disk. Is it something we need to materialize as an output. And i.e. these are our results from the execution run ultimately?